### PR TITLE
fix: group by voucher consolidated showing incorrect data for deferred entries

### PR DESCRIPTION
**Pull request by rohitwaghchaure**

### CODE REVIEW

File: erpnext/accounts/report/general_ledger/general_ledger.py [hunk 1]
```diff
@@ -305,12 +298,22 @@ def update_value_in_dict(data, key, gle):
 			update_value_in_dict(totals, 'total', gle)
 			if filters.get("group_by") != _('Group by Voucher (Consolidated)'):
 				gle_map[gle.get(group_by)].entries.append(gle)
+			elif filters.get("group_by") == _('Group by Voucher (Consolidated)'):
+				key = (gle.get("posting_date"), gle.get("voucher_type"),
```
Comment (nabinhait) [inline erpnext/accounts/report/general_ledger/general_ledger.py]: why posting_date should be part of the key?

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/accounts/report/general_ledger/general_ledger.py (hunk 1 above)
Comment (rohitwaghchaure) [reply erpnext/accounts/report/general_ledger/general_ledger.py]: Fixed

File: erpnext/accounts/report/general_ledger/general_ledger.py [hunk 2]
```diff
@@ -305,12 +298,22 @@ def update_value_in_dict(data, key, gle):
...
 			if filters.get("group_by") != _('Group by Voucher (Consolidated)'):
 				gle_map[gle.get(group_by)].entries.append(gle)
+			elif filters.get("group_by") == _('Group by Voucher (Consolidated)'):
+				key = (gle.get("voucher_type"), gle.get("voucher_no"),
+					gle.get("account"), gle.get("cost_center"))
+				if key not in consolidated_gle:
+					consolidated_gle.setdefault(key, gle)
+				else:
+					update_value_in_dict(consolidated_gle, key, gle)
 			else:
```
Comment (nabinhait) [inline erpnext/accounts/report/general_ledger/general_ledger.py]: no scope of `else` statement

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/public/node_modules [hunk 3]
```diff
@@ -0,0 +1 @@
+/Users/rohitwaghchaure/Documents/erpnext_new/frappe-bench/apps/erpnext/node_modules
```
Comment (nabinhait) [inline erpnext/public/node_modules]: remove this?

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/public/node_modules (hunk 3 above)
Comment (rohitwaghchaure) [reply erpnext/public/node_modules]: Done

**Review by nabinhait — APPROVED** [review]

**Pull request merged**